src/main/java/org/folio/mr/service/impl/MediatedRequestActionsServiceImpl.java

```diff
@@ -191,4 +187,33 @@ private static MediatedRequestWorkflowLog buildMediatedRequestWorkflowLog(
 
     return log;
   }
+
+  @Override
+  public void decline(UUID id) {
+    log.info("decline:: looking for mediated request: {}", id);
+    MediatedRequestEntity mediatedRequest = findMediatedRequest(id);
+    log.debug("decline:: mediatedRequest: {}", mediatedRequest);
+
+    declineRequest(mediatedRequest);
+    mediatedRequestsRepository.save(mediatedRequest);
+    log.info("decline:: mediated request {} was successfully declined", id);
+  }
+
+  private void declineRequest(MediatedRequestEntity request) {
+    if (request.getMediatedRequestStatus() != MediatedRequestStatus.NEW ||
+      !MediatedRequestStep.AWAITING_CONFIRMATION.getValue().equals(request.getMediatedRequestStep()))
+    {
+      throw ExceptionFactory.badRequest("Mediated request status should be 'New - Awaiting conformation'");
```

Comment:
It's ok to throw `ValidationException` here, but by calling a method named `badRequest` we are leaking information about our API into the service, which shouldn't really know that it's a part of any API. Mapping exceptions to response codes is controller's job. We can avoid it by giving this method a more generic name, e.g. "validationError". What do you think?

Also, I don't see a handler for this exception. Does it mean that it will be mapped to a 500 error?

Reply:
I added a handler and yes - better to have http status agnostic service but I'm not sure if it can be *fully* http agnostic.
WDYT about name `unprocessableEntity`?

Reply:
`unprocessableEntity` is still reminiscent of 422 status code, which is not much different from having method called `badRequest`. How about a method `validationError()` which throws a `ValidationException`? IMHO this exception type is generic enough to decouple the service from the API, and we can still legally map it to 422 in our exception handler. BTW, calling such errors "validation error" and mapping them to 422 is common practice is other Folio modules.